test/test_torch.py

```diff
@@ -18021,6 +18020,48 @@ def is_view_of(self, base, other):
 
         return True
 
+    def transpose_if_not_contig(self, x, contiguous=False, dim0=0, dim1=1):
+        if contiguous:
+            return x
+        else:
+            return x.transpose(dim0, dim1)
+
+    @onlyOnCPUAndCUDA
+    def test_view_as_complex(self, device):
+        def fn(contiguous_input=True, dim0=0, dim1=1):
+            t = torch.randn(3, 4, 2, device=device)
+            c_t = t[:, :, 0] + 1j * t[:, :, 1]
+
+            input = self.transpose_if_not_contig(t, contiguous_input, dim0, dim1)
+
+            if (input.size()[-1] != 2 or
```

Comment:
Should also check for the even stride requirement when not the innermost. Also, is that innermost check correct? What if the innermost dim has stride 0?

Reply:
yeah good point! added a case which checks that